File: ASP.NET/Controllers/RestaurantsController.cs
using System;
using System.Web.Mvc;
using BurningPlate.Services;

namespace BurningPlate.Controllers
{
    public class RestaurantsController : Controller
    {
        public RestaurantsController(IRestaurantService service)
        {
            Service = service;            
        }

        protected IRestaurantService Service { get; set; }

        public ActionResult Index()
        {
            ViewData.Model = new RestaurantsViewModel {Title = "Restaurants", Restaurants = Service.All()};

            return View();
        }

        public ActionResult View(Guid? id)
        {
            if (id.HasValue)
            {
                var restaurant = Service.GetById(id.Value);

                if (restaurant != null)
                {
                    ViewData.Model = new RestaurantsViewModel { Title = restaurant.Name, Restaurant = restaurant };

                    return View();
                }   
            }

            ViewData.Model = new RestaurantsViewModel { Title = "Restaurant Not Found"};
            Response.StatusCode = 404;

            return View();
        }
    }
}

File: Rails/app/controllers/restaurants_controller.rb
class RestaurantsController < ApplicationController
  
  def index
    @title = 'Restaurants'
    @restaurants = Restaurant.all
  end
  
  def view
    @restaurant = Restaurant.find_by_id(params[:id])
    
    if @restaurant
      @title = @restaurant.name
    else
      @title = 'Restaurant Not Found'
      render :status => 404  
    end
  end
end
